go-controller/pkg/cni/ovs.go

```diff
@@ -233,20 +238,69 @@ func doPodFlowsExist(queries []openflowQuery) bool {
 	return true
 }
 
-func waitForPodInterface(ctx context.Context, mac string, ifAddrs []*net.IPNet, ifaceName, ifaceID string, ofPort int, checkExternalIDs bool) error {
+// checkCancelSandbox checks that this sandbox is still valid for the current
+// instance of the pod in the apiserver. Sandbox requests and pod instances
+// have a 1:1 relationship determined by pod UID. If we detect that the pod
+// has changed either UID or MAC terminate this sandbox request early instead
+// of waiting for OVN to set up flows that will never exist.
+func checkCancelSandbox(mac string, podLister corev1listers.PodLister, kclient kubernetes.Interface,
+	namespace, name, initialPodUID string) error {
+	pod, err := getPod(podLister, kclient, namespace, name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("pod deleted")
+		}
+		klog.Warningf("[%s/%s] failed to get pod while waiting for OVS port binding: %v", namespace, name, err)
+		return nil
+	}
+
+	if pod == nil {
+		// Not all node CNI modes can pass non-nil podLister or kclient in which
+		// case pod will be nil
+		return nil
+	}
+
+	if string(pod.UID) != initialPodUID {
+		// Pod UID changed and this sandbox should be canceled
+		// so the new pod sandbox can run
+		return fmt.Errorf("canceled old pod sandbox")
+	}
+
+	ovnAnnot, err := util.UnmarshalPodAnnotation(pod.Annotations)
+	if err != nil {
+		return fmt.Errorf("pod OVN annotations deleted or invalid")
```

Comment:
In failure here can we also provide the pod name?

Reply:
@astoycos that's attached by the caller of this function:

https://github.com/ovn-org/ovn-kubernetes/pull/2275/files/53e916b2dbe85fba13fbe8d8a5ba27586aa8ed99#diff-01a2ef868473804cd3d9b894b093e5be0735cfb21bf27f375ece3a33cc07edc9R318